File: Contas/server/Contas.Core/Specifications/Params/CredorParams.cs
using Contas.Core.Specifications.Base;

namespace Contas.Core.Specifications.Params;

public class CredorParams : SpecificationParams
{
    public int? SegmentoDoCredorId { get; set; }
    public string? NomeFantasia { get; set; }
    public string? RazaoSocial { get; set; }
    public long? CNPJ { get; set; }
}

File: Contas/client/src/app/shared/params/credor.params.ts
import { Params } from "@/core/models/params.model";

/**
 * @author Marcelo M. de Castro
 * @summary Parâmetros de consulta para Credor
 * @version 1.0
 */
export class CredorParams extends Params {
    segmentoDoCredorId: number = 0;
    razaoSocial: string = '';
    nomeFantasia: string = '';
    cnpj: number = 0;
}
